sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -2079,4 +2078,601 @@ case class GpuConv(num: Expression, fromBase: Expression, toBase: Expression)
   override def third: Expression = toBase
 
   override def dataType: DataType = StringType
-}
\ No newline at end of file
+}
+
+case class GpuFormatNumber(x: Expression, d: Expression)
+    extends GpuBinaryExpression with ExpectsInputTypes with NullIntolerant {
+  
+  override def left: Expression = x
+  override def right: Expression = d
+  override def dataType: DataType = StringType
+  override def nullable: Boolean = true
+  override def inputTypes: Seq[AbstractDataType] = Seq(NumericType, IntegerType)
+
+  private def stripLeadingZeros(cv: ColumnVector): ColumnVector = {
```

Comment:
Could you add some comments here, because the name of the function appears to be off from what it actually does. Converting 0000 to 0 is fine, but 00010 to 0 is not what I would expect from stripLeadingZeros.